diff -ru a/atk/atkobject.c b/atk/atkobject.c
--- a/atk/atkobject.c	2011-09-19 21:06:54.000000000 +0400
+++ b/atk/atkobject.c	2012-01-28 17:29:07.629332524 +0400
@@ -365,6 +365,7 @@
 
 static HMODULE atk_dll;
 
+#ifdef DLL_EXPORT
 BOOL WINAPI
 DllMain (HINSTANCE hinstDLL,
 	 DWORD     fdwReason,
@@ -379,6 +380,7 @@
 
   return TRUE;
 }
+#endif // DLL_EXPORT
 
 static const char *
 get_atk_locale_dir (void)
