diff -ru a/gdk-pixbuf/gdk-pixbuf-io.c b/gdk-pixbuf/gdk-pixbuf-io.c
--- a/gdk-pixbuf/gdk-pixbuf-io.c	2011-12-17 04:47:20.000000000 +0400
+++ b/gdk-pixbuf/gdk-pixbuf-io.c	2012-01-22 19:00:51.824692680 +0400
@@ -235,6 +235,8 @@
 
 static HMODULE gdk_pixbuf_dll;
 
+#ifdef DLL_EXPORT
+/*
 BOOL WINAPI
 DllMain (HINSTANCE hinstDLL,
          DWORD     fdwReason,
@@ -248,6 +250,8 @@
 
   return TRUE;
 }
+*/
+#endif
 
 char *
 _gdk_pixbuf_win32_get_toplevel (void)

