diff -ru a/src/cairo-system.c b/src/cairo-system.c
--- a/src/cairo-system.c	2010-06-18 15:47:14.000000000 +0400
+++ b/src/cairo-system.c	2012-01-22 20:05:24.524236261 +0400
@@ -69,6 +69,7 @@
 #include <windows.h>
 
 /* declare to avoid "no previous prototype for 'DllMain'" warning */
+/*
 BOOL WINAPI
 DllMain (HINSTANCE hinstDLL,
          DWORD     fdwReason,
@@ -91,7 +92,7 @@
 
     return TRUE;
 }
-
+*/
 #endif
 #endif
 

