diff -ru a/fontconfig.pc.in b/fontconfig.pc.in
--- a/fontconfig.pc.in	2009-11-16 20:24:52.000000000 +0300
+++ b/fontconfig.pc.in	2012-01-26 16:21:16.847331765 +0400
@@ -6,6 +6,7 @@
 Name: Fontconfig
 Description: Font configuration and customization library
 Version: @VERSION@
-Libs: -L${libdir} -lfontconfig
+Requires: freetype2
+Libs: -L${libdir} -lfontconfig -lexpat
 Libs.private: @LIBXML2_LIBS@ @EXPAT_LIBS@ @FREETYPE_LIBS@ @ICONV_LIBS@
 Cflags: -I${includedir}

