diff -ru a/pangocairo.pc.in b/pangocairo.pc.in
--- a/pangocairo.pc.in	2011-08-16 05:11:08.000000000 +0400
+++ b/pangocairo.pc.in	2012-01-25 14:05:49.909263231 +0400
@@ -6,6 +6,6 @@
 Name: Pango Cairo
 Description: Cairo rendering support for Pango
 Version: @VERSION@
-Requires: pango cairo @PKGCONFIG_CAIRO_REQUIRES@
+Requires: pango cairo @PKGCONFIG_CAIRO_REQUIRES@ gthread-2.0
 Libs: -L${libdir} -lpangocairo-@PANGO_API_VERSION@
 Cflags: -I${includedir}/pango-1.0
diff -ru a/pangowin32.pc.in b/pangowin32.pc.in
--- a/pangowin32.pc.in	2011-08-16 05:11:08.000000000 +0400
+++ b/pangowin32.pc.in	2012-01-25 14:25:36.006990705 +0400
@@ -7,5 +7,5 @@
 Description: Win32 GDI font support for Pango
 Version: @VERSION@
 Requires: pango
-Libs: -L${libdir} -lpangowin32-@PANGO_API_VERSION@ @WIN32_LIBS@
+Libs: -L${libdir} -lpangowin32-@PANGO_API_VERSION@ @WIN32_LIBS@ -lusp10
 Cflags: -I${includedir}/pango-1.0

