diff -ru a/gdk-2.0.pc.in b/gdk-2.0.pc.in
--- a/gdk-2.0.pc.in	2011-11-10 21:38:03.000000000 +0400
+++ b/gdk-2.0.pc.in	2012-01-25 14:42:41.276602615 +0400
@@ -8,5 +8,5 @@
 Description: GTK+ Drawing Kit (${target} target)
 Version: @VERSION@
 Requires: @GDK_PACKAGES@
-Libs: -L${libdir} -lgdk-${target}-@GTK_API_VERSION@ @GDK_EXTRA_LIBS@
+Libs: -L${libdir} -lgdk-${target}-@GTK_API_VERSION@ @GDK_EXTRA_LIBS@ -luuid -limm32
 Cflags: -I${includedir}/gtk-2.0 -I${libdir}/gtk-2.0/include @GDK_EXTRA_CFLAGS@
diff -ru a/gtk+-2.0.pc.in b/gtk+-2.0.pc.in
--- a/gtk+-2.0.pc.in	2011-11-10 21:38:03.000000000 +0400
+++ b/gtk+-2.0.pc.in	2012-01-25 14:42:55.955339352 +0400
@@ -11,5 +11,5 @@
 Description: GTK+ Graphical UI Library (${target} target)
 Version: @VERSION@
 Requires: gdk-${target}-2.0 @GTK_PACKAGES@
-Libs: -L${libdir} -lgtk-${target}-@GTK_API_VERSION@ @GTK_EXTRA_LIBS@
+Libs: -L${libdir} -lgtk-${target}-@GTK_API_VERSION@ @GTK_EXTRA_LIBS@ -limm32
 Cflags: -I${includedir}/gtk-2.0 @GTK_EXTRA_CFLAGS@

