diff -ru a/configure.in b/configure.in
--- a/configure.in	2011-11-10 21:38:03.000000000 +0400
+++ b/configure.in	2012-01-28 19:50:46.013914979 +0400
@@ -170,16 +170,16 @@
 CXXFLAGS="$gtk_save_cxxflags"
 AC_LANG_RESTORE
 
-if test "$os_win32" = "yes"; then
-  if test x$enable_static = xyes -o x$enable_static = x; then
-    AC_MSG_WARN([Disabling static library build, must build as DLL on Windows.])
-    enable_static=no
-  fi
-  if test x$enable_shared = xno; then
-    AC_MSG_WARN([Enabling shared library build, must build as DLL on Windows.])
-  fi
-  enable_shared=yes
-fi
+#if test "$os_win32" = "yes"; then
+#  if test x$enable_static = xyes -o x$enable_static = x; then
+#    AC_MSG_WARN([Disabling static library build, must build as DLL on Windows.])
+#    enable_static=no
+#  fi
+#  if test x$enable_shared = xno; then
+#    AC_MSG_WARN([Enabling shared library build, must build as DLL on Windows.])
+#  fi
+#  enable_shared=yes
+#fi
 
 AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL

